remix-analyzer/src/solidity-analyzer/modules/abstractAstView.ts

```diff
@@ -1,15 +1,16 @@
-import { isContractDefinition,  getStateVariableDeclarationsFormContractNode, isInheritanceSpecifier,
-  getInheritsFromName, isFunctionDefinition, isModifierDefinition, isModifierInvocation, getContractName,
-  getFunctionOrModifierDefinitionParameterPart, getType, getDeclaredVariableName, isVariableDeclaration,
-  getFunctionOrModifierDefinitionReturnParameterPart } from './staticAnalysisCommon'
+import { getStateVariableDeclarationsFromContractNode, getInheritsFromName, getContractName,
+  getFunctionOrModifierDefinitionParameterPart, getType, getDeclaredVariableName, getFunctionDefinitionReturnParameterPart } from './staticAnalysisCommon'
 import { AstWalker } from 'remix-astwalker'
+import { FunctionDefinitionAstNode, ParameterListAstNode, ModifierDefinitionAstNode, ContractHLAst, VariableDeclarationAstNode, FunctionHLAst, ContractDefinitionAstNode, ReportObj, ReportFunction, VisitFunction, ModifierHLAst, CompilationResult } from 'types'
+
+type WrapFunction = ((contracts: ContractHLAst[], isSameName: boolean) => ReportObj[])
 
 export default class abstractAstView {
-  contracts = []
-  currentContractIndex = null
-  currentFunctionIndex = null
-  currentModifierIndex = null
-  isFunctionNotModifier = false
+  contracts: ContractHLAst[] = []
+  currentContractIndex: number = -1
+  currentFunctionIndex: number = -1
+  currentModifierIndex: number = -1
+  isFunctionNotModifier: boolean = false
```

Comment:
For "number", "string" & "boolean" you don't need to hardtype the property, it's going to be guessed by typescript

Reply:
ok